validate.py

```diff
@@ -30,9 +31,37 @@ def validate_file(path: Path) -> list[str]:
 
     errors = validate_cross_references(path.name, root)
     errors += validate_multilingual_elements(path.name, root)
+    errors += validate_version_number(path, git_base_ref)
     return errors
 
 
+def validate_version_number(path: Path, git_base_ref: str = None) -> list[str]:
+    """Validates that the version number in the file given by the path `path` has changed, by
+    comparing it to the version number in the same file of the commit given by `git_base_ref`. The
+    validation passes if no `git_base_ref` is given, or if the file has not changed."""
+    if not git_base_ref:
+        return []
+    if not subprocess.check_output(["git", "diff", "--name-only", git_base_ref, "--", path]).strip():
+        return []
+
+    filename = path.name
+    new_root = etree.parse(open(path))
+    new_version = int(new_root.find("version").text)
+
+    try:
+        original_content = subprocess.check_output(["git", "show", f"{git_base_ref}:{path}"], stderr=subprocess.STDOUT)
+        original_root = etree.fromstring(original_content)
+        original_version = int(original_root.find("version").text)
+    except (subprocess.CalledProcessError, etree.XMLSyntaxError, ValueError) as e:
```

Comment:
While probably fine for a little script like this, this is kind of bad error handling. 

Not entirely sure this is worth spending the time on to fix.

Reply:
Yeah, I was originally catching each error type individually at the appropriate line, and decided it wasn't worth the clutter here, almost decided just to let it crash, then thought it might be nice to at least try each file if it happens. In theory they shouldn't happen because the "original" should always be clean (as the current target branch should be already verified).